### PR TITLE
This is implemented by calling getattr(object, name) and seeing wheth…

### PR DESCRIPTION
**Pull request by dmazzella**
…er it raises an AttributeError or not.

### CODE REVIEW

**Comment by dpgeorge** [comment]
The reason this was not fixed before is because 1) no-one needed it; 2) it's expensive (in stack usage) to push an nlr buffer.

Did you find a need for it?

File: py/modbuiltins.c [hunk 1]
```diff
@@ -553,15 +553,24 @@ MP_DEFINE_CONST_FUN_OBJ_3(mp_builtin_setattr_obj, mp_builtin_setattr);
...
+    }
+    else {
+        // an exception was raised
+        mp_obj_type_t *t = (mp_obj_type_t*)((mp_obj_base_t*)nlr.ret_val)->type;
+        if (t == &mp_type_AttributeError) {
+            return mp_obj_new_bool(0);
+        }
+        else {
+            // re-raise exception
+            nlr_jump(nlr.ret_val);
```
Comment (dpgeorge) [inline py/modbuiltins.c]: Please use nlr_raise.  Please provide a test which gets to this line.

**Comment by pfalcon** [comment]
Yep, I'm also -1 on adding this unless shown that it's unavoidable. So far, everything worked well without it.

**Comment by pfalcon** [comment]
Closing for now, please follow recommendations in https://github.com/micropython/micropython/wiki/ContributorGuidelines in the future.

**Pull request closed without merging**